### PR TITLE
Clean projectinfomanager

### PR DESCRIPTION
**Pull request by cartermp**
This is from removing an IVT to FSharp.Compiler.Private.

* Exposes `IsScript` that lives in CompileOps via the compiler service
* Renames the `VersionStamp` type we have in the FSharp.PatternMatcher so it doesn't collide with the `VersionStamp` we reference from Roslyn

### CODE REVIEW

File: vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs [hunk 1]
```diff
@@ -13,16 +13,16 @@ internal class SpellChecker
     {
         private const string SerializationFormat = "2";
 
-        public VersionStamp Version { get; }
+        public PatternMatcherVersionStamp Version { get; }
```
Comment (KevinRansom) [inline vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: For my education why is the rename to PatternMatcherVersionStamp an improvement?

Thanks

Kevin

Comment (KevinRansom) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: Cripes Roslyn's name collides.  That's a bit sucky.

Comment (KevinRansom) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: Although this is a member on a C# instance.  So I don't really see where the collision would occur?

Comment (cartermp) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: It happened in the IDE tools - perhaps it wouldn't have been an issue at build, but the editor was painted red so I just did this and it fixed it.

Comment (KevinRansom) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: Which file was showing the red squiggles, I will take a peek?

Comment (cartermp) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: FSharpProjectOptionsManager.fs

Comment (cartermp) [reply vsintegration/src/FSharp.PatternMatcher/SpellChecker.cs]: As explained yesterday this is an issue in the compiler that causes this, as metadata import order matters, which was causing this clash. Hence the rename.

**Pull request merged**